startup.py

```diff
@@ -23,83 +23,153 @@ class PhotoshopLauncher(SoftwareLauncher):
+                continue
 
-        elif sys.platform == "win32":
-            # Default installs are located here:
-            # C:\program files\Adobe\Adobe Photoshop CC 2017\Photoshop.exe
-            glob_pattern = r"C:\Program Files\Adobe\Adobe Photoshop CC *\Photoshop.exe"
-            version_regex = re.compile(
-                r"^C:\\Program Files\\Adobe\\Adobe Photoshop CC (.+)\\Photoshop.exe$",
-                re.IGNORECASE
-            )
+        return software_versions
+
+    def _find_software_versions(self):
+        """
+        Scan the filesystem for all photoshop executables.
+
+        :return: A list of :class:`SoftwareVersion` objects.
+        """
+
+        self.logger.debug("Scanning for photoshop executables...")
 
-        else:
+        # use the bundled icon
+        icon_path = os.path.join(
+            self.disk_location,
+            "resources",
+            "ps_2017_icon_256.png"
+        )
+        self.logger.debug("Using icon path: %s" % (icon_path,))
+
+        if sys.platform not in self.EXECUTABLE_MATCH_TEMPLATES:
             self.logger.debug("Photoshop not supported on this platform.")
             return []
 
-        self.logger.debug("Scanning for photoshop installations in '%s'..." % glob_pattern)
-        paths = glob.glob(glob_pattern)
+        # all the executable templates for the current OS
+        match_templates = self.EXECUTABLE_MATCH_TEMPLATES[sys.platform]
 
-        software_versions = []
-        for path in paths:
-            # extract version number
-            self.logger.debug("Found photoshop install in '%s'" % path)
-
-            match = version_regex.match(path)
-            if match:
-                # extract first group to get version string
-                dcc_version = match.group(1)
-                self.logger.debug("This is version '%s'" % dcc_version)
-
-                # see if we have a version filter
-                if versions and dcc_version not in versions:
-                    self.logger.debug(
-                        "Skipping this version since it does not match version filter %s" % versions
-                    )
-                elif not self.is_version_supported(dcc_version):
-                    self.logger.info(
-                        "Found Photoshop install in '%s' but only versions %s "
-                        "and above are supported" % (path, self.minimum_supported_version)
-                    )
-                else:
-                    # all good
-                    display_name = "CC %s" % dcc_version
-                    software_version = SoftwareVersion(dcc_version, display_name, path, icon_path)
-                    software_versions.append(software_version)
-            else:
-                self.logger.warning("Could not extract version number from path '%s'" % path)
+        # all the discovered executables
+        all_sw_versions = []
 
-        return software_versions
+        for match_template in match_templates:
```

Comment:
How similar is this to the logic in Houdini? Are you starting to get a feeling that we could refactor this?